coreos-base/coreos-metadata/files/coreos-metadata.service

```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=CoreOS Metadata Agent
+Requires=network-online.target
+After=network-online.target
```

Comment:
- Do all OEMs need networking? If there is a metadata scheme that doesn't need networking then there is no reason to assume the host even has a working network at the moment.
- Our current patched version of networkd-wait-online will block until just one interface is up, this may not be the interface required to fetch metadata so coreos-metadata needs to have a wait-retry loop anyway. If coreos-metadata can handle waiting gracefully there is no point to networkd-wait-online.
- The systemd folks regard networkd-wait-online as a hack, preferring apps to be able to handle the dynamic nature of networking natively and intend networkd-wait-online to only be used when it is impractical to fix an app.